src/test/kotlin/uk/gov/gdx/datashare/integration/repository/AcquirerSubscriptionRepositoryTest.kt

```diff
@@ -77,4 +74,259 @@ class AcquirerSubscriptionRepositoryTest(
     )
       .isEmpty()
   }
+
+  @Test
+  fun `findAllByWhenDeletedIsNull returns the correct values`() {
+    val acquirer = acquirerRepository.save(AcquirerBuilder().build())
+    val returnedSubscription = acquirerSubscriptionRepository.save(
+      AcquirerSubscriptionBuilder(acquirerId = acquirer.id).build(),
+    )
+    val notReturnedSubscription = acquirerSubscriptionRepository.save(
+      AcquirerSubscriptionBuilder(
+        acquirerId = acquirer.id,
+        whenDeleted = LocalDateTime.now(),
+      ).build(),
+    )
+
+    val subscriptions =
+      acquirerSubscriptionRepository.findAllByWhenDeletedIsNull()
+
+    assertThat(
+      subscriptions.filter { s -> s.acquirerSubscriptionId == returnedSubscription.id },
+    )
+      .hasSize(1)
+
+    assertThat(
+      subscriptions.filter { s -> s.acquirerSubscriptionId == notReturnedSubscription.id },
+    )
+      .isEmpty()
+  }
+
+  @Test
+  fun `findAllByOauthClientIdAndWhenDeletedIsNull returns the correct values`() {
+    val acquirer = acquirerRepository.save(AcquirerBuilder().build())
+    val returnedSubscription = acquirerSubscriptionRepository.save(
+      AcquirerSubscriptionBuilder(
+        acquirerId = acquirer.id,
+        oauthClientId = "test",
+      ).build(),
+    )
+    val whenDeletedNotNullSubscription = acquirerSubscriptionRepository.save(
+      AcquirerSubscriptionBuilder(
+        acquirerId = acquirer.id,
+        oauthClientId = "test",
+        whenDeleted = LocalDateTime.now(),
+      ).build(),
+    )
+    val incorrectClientIdSubscription = acquirerSubscriptionRepository.save(
+      AcquirerSubscriptionBuilder(
+        acquirerId = acquirer.id,
+        oauthClientId = null,
+      ).build(),
+    )
+
+    val subscriptions =
+      acquirerSubscriptionRepository.findAllByOauthClientIdAndWhenDeletedIsNull("test")
+
+    assertThat(
+      subscriptions.filter { s -> s.acquirerSubscriptionId == returnedSubscription.id },
+    )
+      .hasSize(1)
+
+    assertThat(
+      subscriptions.filter { s -> s.acquirerSubscriptionId == whenDeletedNotNullSubscription.id || s.acquirerSubscriptionId == incorrectClientIdSubscription.id },
+    )
```

Comment:
Do we want these in separate assertions?

Reply:
I would say yes pls, so that if it fails we know which case does